runners/direct-java/src/main/java/org/apache/beam/runners/direct/DoFnLifecycleManagerRemovingTransformEvaluator.java

```diff
@@ -53,6 +55,15 @@ public void processElement(WindowedValue<InputT> element) throws Exception {
     }
   }
 
+  public void onTimer(TimerData timer, BoundedWindow window) throws Exception {
```

Comment:
There probably should be a JIRA to make this consistent with `SplittableDoFn` and `GroupAlsoByWindow`